### DISCUSSION TITLE
[FeatureRequest]: Graph x-axis support for multiple timezones

### DISCUSSION THREAD

**Discussion by ShawnTuatara** [grafana/grafana#38237] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Looking back on other issues raised about timezones seems to be how to show the graph time axis in a way that is easy to interpret by users from potentially disparate locations. I appreciate the option to have the dashboard displaying everything in browser timezone or UTC timezone (which is similar to other applications like Datadog and Kibana). The difficulty is that there is no way to satisfy each user of the dashboard as some work with UTC while others prefer working locally.

My request is to have the option to show, at a minimum, two timezone times on the x-axis. That way it is straight forward to the viewers of the graph and to others that the graph is shared with as to the time that is being displayed.

![grafana-dual-timezone](https://user-images.githubusercontent.com/326589/110602679-175daa00-813b-11eb-9e71-123cde19ac1d.png)
UTC being displayed nearest to the axis and UTC-8 (browser timezone) below it.

This could be set either on the specific graph by having an optional "secondary timezone" option:
![image](https://user-images.githubusercontent.com/326589/110603067-7a4f4100-813b-11eb-86dc-7bb7dffa298a.png)

Or on the dashboard settings to specify the "secondary timezone":
![image](https://user-images.githubusercontent.com/326589/110603162-905d0180-813b-11eb-976f-c461fb2a56e9.png)

**Why is this needed**:
Reduce the burden on the end users of the graph to figure out which timezone is being used and then having to convert that timezone based on other factors.

**Comment by leeoniya** [comment]
there's a lot of trickiness in the details here. for example, let's say you have UTC primary and UTC-6 secondary. the tick locations will be chosen to align nicely with the UTC primary, like Jan 1, 2021 00:00:00. if you're zoomed out, this tick will show just the Jan 2021, good so far. now what will the UTC-6 show below it? Dec 2020? that's accurate but probably confusing.

> The difficulty is that there is no way to satisfy each user of the dashboard as some work with UTC while others prefer working locally.

i think we need the possibility to decouple the saved dashboard timezone and the user's-preferred timezone, which can be stored in a cookie. we can query the user's browser timezone and offer to store an alternate tz preference if it does not match the saved dash timezone.

cc @hugohaggmark

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
